### ISSUE TITLE
Ons-pull-hook: threshold-height

### ISSUE THREAD

**Issue by Hirbod** [OnsenUI/OnsenUI#2202]
Hi,
https://onsen.io/v2/api/js/ons-pull-hook.html

states that threshold-height will be disabled, if smaller then "height". This is actually nonsense. I know what the "thinking" was behind of it but there are many visual usecases where you want threshold-height to be smaller then height. You ask why?

Well, cause I use ons-pull-hook for a "curtain" effect, not actually for reloading any content. I got it working perfect now, but I had to duplicate the effect under a new name for me and change

```js
    key: '_thresholdHeightEnabled',
    value: function _thresholdHeightEnabled() {
      var th = this.thresholdHeight;
      return th > 0 && th >= this.height;
    }
```

to

```js
    key: '_thresholdHeightEnabled',
    value: function _thresholdHeightEnabled() {
      var th = this.thresholdHeight;
      return th > 0;
    }
```

I think this should be the default setting as this will make the ons-pull-hook usable for other scenarios (like an top off-canvas menu) and will prevent an "cutted element" after x height, because the threshold triggers before height.

To make this more understandable, I've uploaded a video.
https://www.youtube.com/watch?v=kIRd7Thx43I

As you can see inside the video, there is also a small moment of "glitch", which I am still trying to debug (as of momentum/intertial scroll).

Long story short: there is no reason to trigger thresholdHeight only when >= height.
Please remove this, so people have the ability to use this awesome directive in other ways (like me)

**Comment by frandiox** [comment]
@Hirbod Hi! That implies that the pull hook would enter `action` state without `preaction` first, right? Not a huge deal but perhaps there's another way to do what you want. Have you tried the `onPull` property? I think you can do the same and perhaps even more with it.

**Comment by Hirbod** [comment]
@frandiox you are right with everything. I did not try onPull. Do I have to pass it as 
```
<ons-pull-hook on-pull="doSomthing">
```

?

**Comment by Hirbod** [comment]
@frandiox on-pull never get called. I have 2.6.0 installed.. I am never getting a call on my defined function, while "action" works.

**Comment by frandiox** [comment]
@Hirbod It's a DOM property, `document.querySelector('ons-pull-hook').onPull = function() {};`. There is an example (with Vue) [here](https://github.com/OnsenUI/vue-onsenui-kitchensink/blob/master/src/pages/PullHook.vue#L9).
I'm not 100% sure if you can use it for your specific usecase though. I'll consider with the team about removing the `th >= this.height`, although this is not a normal `ons-pull-hook` use case I guess.

**Comment by PeaFull** [comment]
@frandiox Hi! How should onPull be used in angular 1？

**Comment by PeaFull** [comment]
@frandiox What events can be monitored at the beginning of the pull?

**Comment by Hirbod** [comment]
Hi @frandiox 

any news regarding removing the linke?

**Comment by frandiox** [comment]
@Hirbod Merged in master!

@PeaFull No events at the very beginning, only `onPull` will be called. In 2.8.3 you can access `onPull` like `$scope.myPullHook.onPull` (and seeting `<ons-pull-hook var="myPullHook">`, of course). Keep in mind that directives in AngularJS are linked to their scope in a specific order parent > child, so the parent controller might run before `$scope.myPullHook` is set. You can wait for page's `init` event to ensure the `$scope` is populated.

**Comment by PeaFull** [comment]
@frandiox Thanks for your reply.

**Comment by Hirbod** [comment]
@frandiox thanks you mate!
